pkg/p2p/kadcli/reader.go

```diff
@@ -0,0 +1,120 @@
+// This Source Code Form is subject to the terms of the MIT License.
+// If a copy of the MIT License was not distributed with this
+// file, you can obtain one at https://opensource.org/licenses/MIT.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+package kadcli
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"io"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/peer"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/checksum"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/protocol"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/topics"
+	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus"
+	"github.com/dusk-network/dusk-protobuf/autogen/go/rusk"
+)
+
+// Reader abstracts all of the logic and fields needed to receive messages from
+// other network nodes.
+type Reader struct {
+	processor *peer.MessageProcessor
+	gossip    *protocol.Gossip
+
+	cli  rusk.NetworkClient
```

Comment:
Please rename this to `client` as cli might be confused with CLI (Command Line Interface). Once we deprecate the `kadcast` golang impl, we'll rename the package from kadcli to `kadcast`